pkg/fatfs/fatfs_vfs/fatfs_vfs.c

```diff
@@ -47,6 +58,27 @@ static void _build_abs_path(fatfs_desc_t *fs_desc, const char *name)
              fs_desc->vol_idx, name);
 }
 
+#ifdef MODULE_FATFS_VFS_FORMAT
+static int _format(vfs_mount_t *mountp)
+{
+    fatfs_desc_t *fs_desc = (fatfs_desc_t *)mountp->private_data;
```

Comment:
```suggestion
    fatfs_desc_t *fs_desc = mountp->private_data;
```

`void *` can be implicitly casted to any other pointer. (And should, as explicit casts turn off a lot of compiler diagnostics.)